### PR TITLE
Fix coding standards

### PR DESCRIPTION
**Pull request by stof**
| Q             | A
| ------------- | ---
| Branch?       | 2.8
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes  
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

This PR is mostly about running the PHP-CS-Fixer (v2.12.1) in the whole codebase.

- I updated the exclude rule to avoid some false positives for the `error_suppression` fixer (we have more files triggering unsilenced deprecations on purpose than when building the initial whitelist, mostly).
- I ran the fixer with this updated config. Most changes were related to fully-qualifying some constants, with the new fixer implemented in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3127, for which @nicolas-grekas and I suggested a config to include in the Symfony ruleset. Based on the output, I suggested a feature request in https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/3872 as we might want to avoid the `\` in non-namespaced files to improve readability. We might want to remove the second commit of this PR if we decide to wait for the feature to be implemented (update: implementation is contributed in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3876)
- I added the `native_function_invocation` fixer explicitly, to automatically fully-qualify calls to compiler-optimized functions. This feature was implemented in PHP-CS-Fixer based on our feature request (as currently, we do such thing only manually in some hot path, because it could not be automated). I opened https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3873 to include it in the ruleset automatically.

TODOs:
- [x] agree on the updated rules
- [x] update fabbot to use the new version of PHP-CS-Fixer
- [x] make separate PRs for newer branches with their own updates (exclude rules, and CS fixes), once this PR gets merged.

### CODE REVIEW

**Comment by keradus** [comment]
@stof , would be great if you can create a benchmark without and with those changes !

**Comment by stof** [comment]
I don't think a benchmark will show much differences here, as the hot path of the DI component (and of a few other places) were already fully-qualified manually by @nicolas-grekas (well, maybe not in 2.8, but at least in newer versions).

**Comment by nicolas-grekas** [comment]
> make separate PRs for newer branches with their own updates (exclude rules, and CS fixes), once this PR gets merged.

Actually, the process for this is:
- merge everything up to master
- merge this PR on 2.8
- merge everything up to master. Resolving conflicts just means running the command again on each branch (thus no need for per-branches PRs.)

**Comment by stof** [comment]
@nicolas-grekas the update of the config file (exclude rules) will be needed for newer branches too (due to renaming some code, or adding more places needing to be excluded).

**Comment by stof** [comment]
but yes, fixing conflicts would be done by running the command again.

**Comment by iltar** [comment]
Wouldn't it make more sense to have `use` statements for global imports? Especially as readability and merge conflicts are a commonly provided rejection for doing this in existing code.

**Comment by stof** [comment]
@iltar this would require using `use const` and `use function`, which cannot be used in 2.8 or in 3.x due to our PHP requirements.

And for classes, our existing rule is already to avoid use statements for global classes, so this is consistent.

and regarding merge conflicts, this is precisely the reason why I'm doing this in 2.8, not in master (so that all branches are using the same CS), and why I'm doing it now and not 6 months ago (the change is now automated)

**Comment by iltar** [comment]
I mention this because a lot of people are complaining about the readability of backslashes in the code, and are asking for imports instead. I do agree with them on this point, because it _does_ hurt readability quite some. I understand we can't merge this in 2.8, but why not do it in 3.4+ only? 2.8 support won't last long anymore either way and to be fair, people should've already upgraded a while ago.

**Comment by stof** [comment]
@iltar 3.4 supports PHP 5.5.9+, so this would not be 3.4+, but 4.0+. And having CS violations in 4.x for any code contributed in 3.4 and merged up would mean that we suffer from branch merging for the whole 3.4 lifetime (i.e. for the next 2 years: https://symfony.com/roadmap/3.4).

**Comment by iltar** [comment]
Ah my bad, I forgot that 3.4 was not on 7.x yet. Yay for LTS.

**Comment by stof** [comment]
regarding the constant invocation, I will revert the change, and disable the rule temporarily, until https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3876 is merged and released. This would avoid some of the existing changes which add visual clutter for no performance benefit.

**Comment by fabpot** [comment]
@stof Can you revert what needs to be reverted so that I can merge this one and then I will take care of propagating changes up to master?

**Comment by fabpot** [comment]
fabbot is now using v2.12.2

**Comment by stof** [comment]
@fabpot done.

The list of excluded files in the php-cs--fixer configuration file (updated in the first commit of this PR) will need some adjustments for newer branches (I have a branch locally with a similar commit for master (I started there before switching target branch), and there were a few differences.
I could perform the propagation of changes to newer branches, taking care of this if you want.

I intentionally kept multiple commits in the history, to better see the different changes performed in this PR (as some of them are automated).

**Comment by nicolas-grekas** [comment]
CliDumperTest.php needs a tweak: the last frame (`%d. %slosure%s() ==> Twig%cTemplate->render(): {`) is now removed on PHP7 & HHVM.

**Comment by nicolas-grekas** [comment]
Let's replace it with a "%A" in the fixture.

File: src/Symfony/Component/Process/Tests/PipeStdinInStdoutStdErrStreamSelect.php [hunk 1]
```diff
@@ -35,22 +35,22 @@
     }
 
     if (in_array(STDOUT, $w) && strlen($out) > 0) {
-        $written = fwrite(STDOUT, (binary) $out, 32768);
+        $written = fwrite(STDOUT, (string) $out, 32768);
```
Comment (keradus) [inline src/Symfony/Component/Process/Tests/PipeStdinInStdoutStdErrStreamSelect.php]: wow, I did not expect to see binary casting so much in real code :D

**Comment by nicolas-grekas** [comment]
Thank you @stof.

**Comment by nicolas-grekas** [comment]
Now merged up to master.
I also enabled the "ordered_imports" fixer.
Thank you all for this!

**Comment by SpacePossum** [comment]
nice to see the new fixers being used :D  thanks all!

**Pull request merged**